compiler/wasm/src/types/noir_artifact.ts

```diff
@@ -58,6 +58,8 @@ export interface ABIVariable {
 export interface NoirFunctionEntry {
   /** The name of the function. */
   name: string;
+  /** The hash of the circuit. */
+  hash?: number | string;
```

Comment:
I think we can break backwards compatibility here without much issues. The cases where we have artifacts used across nargo builds currently is very small.